### PR TITLE
MEG: Add git pre-commit configuration

### PR DESCRIPTION
**Pull request by hoechenberger**
I copied MNE's pre-commit config

Once merged, we can enable pre-commit CI, https://pre-commit.ci

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - id: black
+        args: [--quiet]
+
+  # Ruff mne
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.0.291
+    hooks:
+      - id: ruff
+        name: ruff mne
+        files: ^mne/
```
Comment (larsoner) [inline .pre-commit-config.yaml:14]: Needs to be updated

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  # Ruff tutorials and examples
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.0.291
+    hooks:
+      - id: ruff
+        name: ruff tutorials and examples
```
Comment (larsoner) [inline .pre-commit-config.yaml:21]: Same

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        # D103: missing docstring in public function
+        # D400: docstring first line must end with period
+        args: ["--ignore=D103,D400"]
+        files: ^tutorials/|^examples/
```
Comment (larsoner) [inline .pre-commit-config.yaml:25]: Etc

**Review by larsoner — COMMENTED** [review]
I enabled the GitHub app so if you close this PR and open another it might show up

**Comment by hoechenberger** [comment]
> I enabled the GitHub app so if you close this PR and open another it might show up

Cool, thanks!

**Comment by hoechenberger** [comment]
Oh the pre-commit CI bot is already here :)

**Comment by hoechenberger** [comment]
I removed yamllint and codespell for now, we can add them later

**Comment by hoechenberger** [comment]
I would say this is ready to go. 

In follow-up PRs we should

- Drop flake8
- Add yamllint 
- Add codespell
- Move from setup.cfg to pyproject.toml, which was added in this PR

**Review by sappelhoff — COMMENTED** [review]
Thanks! could you also add some instructions on the pre-commit hooks to the dev documentation, please?

**Comment by hoechenberger** [comment]
@sappelhoff Happy? :)

**Review by sappelhoff — APPROVED** [review]
thanks Richard!

**Pull request merged**